lib/GNUMakefilePyGeNN

```diff
@@ -0,0 +1,136 @@
+endif
+ifeq ($(strip $(PYTHON_NUMPY_INCLUDE)),)
+$(error attempt to automatically detect python numpy package include path failed. Try specifying the path as PYTHON_NUMPY_INCLUDE=)
+endif
+# PYGENN link flags
+ifndef CPU_ONLY
+    ifeq ($(DARWIN),DARWIN)
+        PYGENN_LINK_FLAGS       =-rpath $(CUDA_PATH)/lib -L"$(CUDA_PATH)/lib" -lcuda -lcudart -lstdc++ -lc++
+    else
+        ifeq ($(OS_SIZE),32)
+            PYGENN_LINK_FLAGS   =-L"$(CUDA_PATH)/lib" -lcuda -lcudart
+        else
+            PYGENN_LINK_FLAGS   =-L"$(CUDA_PATH)/lib64" -lcuda -lcudart
+        endif
+    endif
+else
+    SWIG_CPU                    :=-DCPU_ONLY
+    PYGENN_PATH                 :=$(addsuffix _CPU,$(PYGENN_PATH))
+    PYGENN_LINK_FLAGS           :=
+    ifeq ($(DARWIN),DARWIN)
+        PYGENN_LINK_FLAGS       :=-lstdc++ -lc++
+    endif
+endif
+
+$(info $$PYGENN_PATH = ${PYGENN_PATH})
+
+ifdef MPI_ENABLE
+    PYGENN_LINK_FLAGS           +=$(shell mpiCC -showme:link)
+endif
+
+PYGENN_PIC_OBJ                     :=global modelSpec neuronGroup synapseGroup currentSource initVarSnippet utils codeStream codeGenUtils sparseUtils neuronModels postSynapseModels synapseModels newNeuronModels newPostsynapticModels newWeightUpdateModels currentSourceModels standardSubstitutions standardGeneratedSections generateALL generateCPU generateInit generateKernels generateMPI generateRunner
+PYGENN_PIC_OBJ                     :=$(addprefix $(LIBGENN_OBJ_PATH)/,$(PYGENN_PIC_OBJ))
+PYGENN_PIC_OBJ                     :=$(addsuffix _PIC.o,$(PYGENN_PIC_OBJ))
+
+PYGENN_CUSTOM_OBJ                  :=newNeuronModelsCustom newPostsynapticModelsCustom newWeightUpdateModelsCustom currentSourceModelsCustom initVarSnippetCustom
+PYGENN_CUSTOM_OBJ                  :=$(addprefix $(LIBGENN_OBJ_PATH)/,$(PYGENN_CUSTOM_OBJ))
+PYGENN_CUSTOM_OBJ                  :=$(addsuffix _PIC.o,$(PYGENN_CUSTOM_OBJ))
+
+SWIG_SUBMODULES                    :=Snippet InitVarSnippet NewModels NeuronModels PostsynapticModels WeightUpdateModels CurrentSourceModels StlContainers SharedLibraryModel
+SWIG_SUBMODULES                    :=$(addprefix $(PYGENN_PATH)/_,$(SWIG_SUBMODULES))
+SWIG_SUBMODULES                    :=$(addsuffix .so,$(SWIG_SUBMODULES))
+
+pygenn:
+	$(MAKE) pygenn_pre
+	$(MAKE) pygenn_post
+
+pygenn_pre: prepare_swig_config $(PYGENN_PATH) $(LIBGENN_OBJ_PATH) $(PYGENN_PIC_OBJ)
+
+pygenn_post: $(PYGENN_CUSTOM_OBJ) $(SWIG_SUBMODULES)
+	$(SWIG) $(SWIG_CPU) -outdir $(PYGENN_PATH) -o $(PYGENN_PATH)/$(SWIG_MODULE_NAME)_wrap.cc $(PYGENN_SWIG_FILE_PATH)
+	$(CXX) $(CXXFLAGS) -shared -fPIC -DSWIG $(PYGENN_PATH)/$(SWIG_MODULE_NAME)_wrap.cc -o $(PYGENN_PATH)/_$(SWIG_MODULE_NAME).so $(PYGENN_PIC_OBJ) $(PYGENN_CUSTOM_OBJ) $(INCLUDE_FLAGS) -I $(PYTHON_INCLUDE) -I $(SWIG_PATH) $(PYGENN_LINK_FLAGS)
+	cp $(SWIG_PATH)/GeNNModel.py $(PYGENN_PATH)/GeNNModel.py
+	cp $(SWIG_PATH)/GeNNGroups.py $(PYGENN_PATH)/GeNNGroups.py
+
+$(PYGENN_PATH)/_StlContainers.so: $(PYGENN_PATH)/stl_containers_wrap.cc
```

Comment:
It would be really good if (after you include GNUMakefileLibGeNN :smile:) you used the ``LIBGENN_PREFIX`` to make a unique object directory for each variant of the modules. Otherwise it's very easy to get into a mess of incompatible bits of CPU_ONLY and normal versions mixed together.

Reply:
I will try to adapt both make files to minimize redundancy